subjects/devops/deep-in-system/audit/README.md

```diff
@@ -0,0 +1,366 @@
+```console
+user:~$ lsblk -o NAME,FSTYPE,SIZE,MOUNTPOINT /dev/sda
+NAME   FSTYPE SIZE MOUNTPOINT
+sda            30G
+├─sda<...>          1M
+├─sda<...> swap     4G [SWAP]
+├─sda<...> ext4    15G /
+├─sda<...> ext4     5G /home
+└─sda<...> ext4     6G /backup
+user:~$
+```
+- The VM disk size must be 30GB.
+
+- VM disk must be divided into these partitions:
+"swap:" 4G
+"/": 15G
+"/home": 5G
+"/backup": 6G
+
+###### Is the VM Disk size correct?
+> There is no problem if the size of the divisions is not very accurate (Authorized error rate: <= 0.5G)!
+###### Are the VM disk partitions correct?
+
+##### Check the hostname and user name
+To check the hostname:
+```console
+user:~$ hostname
+<username>-host
+user:~$
+```
+To check the user name and groups:
+```console
+user:~$ id
+uid=<...>({username}) gid=<...>({username}) groups=<...>({username}),<...>(sudo),<...>
+user:~$
+```
+###### Does the hostname in the format of "{username}-host"?
+###### Does the student use a user different from the "root" user?
+###### Does the username contain the student login?
+###### Does the user in the sudo group?
+###### Does the student can explain what is sudo group in Linux?
+
+#### The Network & Security Part:
+##### Check the VM IP address
+The student must show the file that was modified to set a static IP address.
+###### Does the student can explain the configuration?
+###### Does the student What is a netmask?
+You can check if the IP address is static with this command:
+```console
+user:~$ ip a | grep dynamic
+user:~$
+```
+###### There is no internet interface with dynamic IP assignment?
+You can check if the internet works fine with the static IP address:
+```console
+user:~$ ping -c 5 google.com
+```
+###### Can connect to the internet properly?
+###### Can The student explain why a static IP address is important for a web server?
+
+##### Check the sshd configuration
+The student must show the file that was modified to secure the ssh server.
+###### Does the student can explain the configuration?
+###### Is the root access disabled in the sshd config (PermitRootLogin: no)?
+###### Is the port of the sshd "2222"
+You can try to connect from outside the VM
+```console
+outsideTheVM:~$ ssh {username}@{machine-ip} -p 2222
+{username}@{machine-ip}'s password:
+Welcome to Ubuntu <......>
+InsideTheVM:~$ hostname
+{username}-host
+InsideTheVM:~$
+```
+###### Can connect to the ssh properly?
+###### Does the student can explain what is ssh server and what the role of it?
+
+##### Check the firewall
+If the student uses ufw you can check it with this command:
+```console
+user:~$ sudo ufw status
+Status: active
+
+To                         Action      From
+--                         ------      ----
+2222/tcp                   ALLOW       Anywhere
+<...>
+Apache                     ALLOW       Anywhere
+<...>
+user:~$
+```
+Otherwise, the student must show what firewall is used.
+###### Is the firewall activated?
+The student must justify all open ports.
```

Comment:
h5.

Ask the student to justify why each open port is open

Reply:
fixed